PublishedModules/Psychokiller1888/Minigames/Minigames.py

```diff
@@ -65,65 +64,77 @@ def onUserCancel(self, session: DialogSession):
 		if self._minigame:
 			self._minigame.started = False
 
+	def minigameIntent(self, intent: str, session: DialogSession) -> bool:
+		self._minigame.onMessage(intent, session)
+		return True
 
-	def onMessage(self, intent: str, session: DialogSession) -> bool:
+	def playGameIntent(self, intent: str, session: DialogSession) -> bool:
 		sessionId = session.sessionId
 		slots = session.slots
 
 		if not self._minigame or not self._minigame.started:
-			if intent == self._INTENT_PLAY_GAME or (intent == self._INTENT_ANSWER_MINI_GAME and session.previousIntent == self._INTENT_PLAY_GAME):
-				if 'WhichGame' not in slots.keys():
-					self.continueDialog(
-						sessionId=sessionId,
-						intentFilter=[self._INTENT_ANSWER_MINI_GAME],
-						text=self.TalkManager.randomTalk('whichGame'),
-						previousIntent=self._INTENT_PLAY_GAME
-					)
-
-				elif session.slotValue('WhichGame') not in self._SUPPORTED_GAMES:
-					self.continueDialog(
-						sessionId=sessionId,
-						intentFilter=[self._INTENT_ANSWER_MINI_GAME, self._INTENT_ANSWER_YES_OR_NO],
-						text=self.TalkManager.randomTalk('unknownGame'),
-						previousIntent=self._INTENT_PLAY_GAME
-					)
-
-				else:
-					game = session.slotValue('WhichGame')
-
-					self._minigame = self._minigames[game]
-					self._minigame.start(session)
-
-			elif intent == self._INTENT_ANSWER_YES_OR_NO:
-				if not commons.isYes(session):
-					self.endDialog(
-						sessionId=sessionId,
-						text=self.randomTalk('endPlaying')
-					)
-				else:
-					self.continueDialog(
-						sessionId=sessionId,
-						intentFilter=[self._INTENT_ANSWER_MINI_GAME],
-						text=self.TalkManager.randomTalk('whichGame'),
-						previousIntent=self._INTENT_PLAY_GAME
-					)
+			if 'WhichGame' not in slots:
+				self.continueDialog(
+					sessionId=sessionId,
+					intentFilter=[self._INTENT_ANSWER_MINI_GAME],
+					text=self.TalkManager.randomTalk('whichGame'),
+					previousIntent=self._INTENT_PLAY_GAME
+				)
+
+			elif session.slotValue('WhichGame') not in self._SUPPORTED_GAMES:
+				self.continueDialog(
+					sessionId=sessionId,
+					intentFilter=[self._INTENT_ANSWER_MINI_GAME, self._INTENT_ANSWER_YES_OR_NO],
+					text=self.TalkManager.randomTalk('unknownGame'),
+					previousIntent=self._INTENT_PLAY_GAME
+				)
 
-		elif self._minigame is not None:
-			if intent == self._INTENT_ANSWER_YES_OR_NO and session.customData and 'askRetry' in session.customData.keys():
-				if commons.isYes(session):
-					self._minigame.start(session)
-				else:
-					self._minigame = None
-					self.endDialog(
-						sessionId=sessionId,
-						text=self.randomTalk('endPlaying')
-					)
 			else:
-				self._minigame.onMessage(intent, session)
+				game = session.slotValue('WhichGame')
+				self._minigame = self._minigames[game]
+				self._minigame.start(session)
 
+		elif self._minigame is not None:
+			self._minigame.onMessage(intent, session)
```

Comment:
can this intent even occur while a game is running? as far as I did see it should always use a intent filter that does not allow this

Reply:
urghh kind of ugly that they show removed files in the review